grafast/grafserv/src/servers/whatwg-node-server/index.ts

```diff
@@ -0,0 +1,154 @@
+import { createServerAdapter } from '@whatwg-node/server'
+
+import { GrafservBase } from "../../core/base.js";
+import type {
+  GrafservBody,
+  GrafservConfig,
+  RequestDigest,
+  Result,
+} from "../../interfaces.js";
+
+import { OptionsFromConfig } from '../../options.js';
+import { httpError } from '../../utils.js';
+
+export function getBodyFromRequest(
+  req: Request /* IncomingMessage */,
+  maxLength: number,
+): Promise<GrafservBody> {
+  return new Promise(async (resolve, reject) => {
+    const chunks: Buffer[] = [];
+    let len = 0;
+    const handleDataCb = (chunk: Uint8Array<ArrayBufferLike>) => {
+      chunks.push(Buffer.from(chunk));
+      len += chunk.length;
+      if (len > maxLength) {
+        reject(httpError(413, "Too much data"));
+      }
+    };
+    const doneCb = () => {
+      resolve({ type: "buffer", buffer: Buffer.concat(chunks) });
+    };
+    const reader = req.body?.getReader()
+    if (!reader) {
+      return doneCb()
+    }
+    while (true) {
+      const {done, value} = await reader?.read()
+      if (value) {
+        handleDataCb(value)
+      }
+      if (done) {
+        return doneCb()
+      }
+    }
+  });
+}
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace Grafast {
+    interface RequestContext {
+        whatwg: {
+            version:string
```

Comment:
Sorry; I meant that the version should be part of the scope key like with the other servers; then when a new version is supported we can add support for that without breaking existing users. See the Koa, Express, Fastify, Lambda and H3 adaptors for example.

```suggestion
        whatwgv0_9: {
```

Reply:
This should also be factored into the file name; e.g. `src/servers/whatwg-node-server/v0_9/...` - see the same list of other adaptors for examples.